lms/djangoapps/verify_student/models.py

```diff
@@ -1186,3 +1188,62 @@ def check_user_skipped_reverification_exists(cls, user, course_id):
             Boolean
         """
         return cls.objects.filter(user=user, course_id=course_id).exists()
+
+
+class CreditCourse(models.Model):
+    """Model for tracking the credit course."""
+
+    course_key = CourseKeyField(max_length=255, db_index=True, unique=True)
+    enabled = models.BooleanField(default=False)
+
+
+class CreditProvider(TimeStampedModel):
+    """This model represents an institution that can grant credit for a course.
+    each provider is identified by unique id e.g ASU
+    """
+    provider_id = models.CharField(max_length=255, db_index=True, unique=True)
+    display_name = models.CharField(max_length=255)
+
+
+class CreditRequirement(TimeStampedModel):
+    """This model represents a credit requirement.
+
+    Each requirement is uniquely identified by a `namespace` and a `name`. CreditRequirements
+    also include a `configuration` dictionary, the format of which varies by the type of requirement.
+    The configuration dictionary provides additional information clients may need to determine
+    whether a user has satisfied the requirement.
+    """
+
+    course = models.ForeignKey(CreditCourse, related_name="credit_requirement_course")
+    namespace = models.CharField(max_length=255)
+    name = models.CharField(max_length=255)
+    configuration = JSONField()
+    active = models.BooleanField(default=True)
+
+    class Meta:  # pylint: disable=missing-docstring, old-style-class
+        unique_together = (('namespace', 'name', 'course'),)
+
+
+class CreditRequirementStatus(TimeStampedModel):
+    """This model represents the status of each requirement."""
+
+    REQUIREMENT_STATUS_CHOICES = (
+        ("satisfied", "satisfied"),
+    )
+
+    username = models.CharField(max_length=255, db_index=True)
+    requirement = models.ForeignKey(CreditRequirement, related_name="status")
+    status = models.CharField(choices=REQUIREMENT_STATUS_CHOICES, max_length=32)
+
+
+class CreditEligibility(TimeStampedModel):
+    """A record of a user's eligibility for credit from a specific credit
+    provider for a specific course.
+    """
+
+    username = models.CharField(max_length=255, db_index=True)
+    course = models.ForeignKey(CreditCourse, related_name="credit_eligibility")
+    provider = models.ForeignKey(CreditProvider, related_name="credit_eligibility")
```

Comment:
Since a foreign key represents a many-to-one relationship, you could further refine this `related_name` by making it plural and more concise: `eligibilities`. A `CreditCourse` or `CreditProvider` will be related to many `CreditEligibility ` records.